---
title: "Analysis"
author: "Sam Gutmanis"
date: "`r Sys.Date()`"
output: html_document
---
#PART 1
Load libraries
```{r}
library(knitr)
library(dplyr)
library(BiocManager)
library(annotate)
```

Import the Sequences.csv file
```{r}
Sequences<- read.csv("./Sequences.csv")
```


Print out each sequence
```{r}
print(Sequences$Sequence)
```
Print out the number of each nucleotide as a table for each of the three sequences.
```{r}
counts <- matrix(0, nrow = 3, ncol = 4, dimnames = list(c("Sequence 1", "Sequence 2", "Sequence 3"), c("A", "T", "C", "G")))

# iterate over each sequence
for (i in 1:length(Sequences$Sequence)) {
  # get the sequence and remove any non-ATCG characters
  seq <- gsub("[^ATCG]", "", Sequences$Sequence[i])
  
  # count the occurrences of each base pair
  counts[i, "A"] <- nchar(gsub("[^A]", "", seq))
  counts[i, "T"] <- nchar(gsub("[^T]", "", seq))
  counts[i, "C"] <- nchar(gsub("[^C]", "", seq))
  counts[i, "G"] <- nchar(gsub("[^G]", "", seq))
}
counts
```

Print the counts for each base of each sequence
```{r}
for (i in 1:nrow(Sequences)) {
  
  # Get the nucleotide sequence for the current sequence
  seq <- Sequences$Sequence[i]
  
  # Initialize counters for each nucleotide
  a_count <- 0
  t_count <- 0
  c_count <- 0
  g_count <- 0
  
  # Loop through each character in the sequence
  for (j in 1:nchar(seq)) {
    
    # Get the current character
    char <- substr(seq, j, j)
    
    # Increment the corresponding counter
    if (char == "A") {
      a_count <- a_count + 1
    } else if (char == "T") {
      t_count <- t_count + 1
    } else if (char == "C") {
      c_count <- c_count + 1
    } else if (char == "G") {
      g_count <- g_count + 1
    }
  }
  
  # Print the results for the current sequence
  cat(sprintf("Sequence %d:\n", i))
  cat(sprintf("A: %d\n", a_count))
  cat(sprintf("T: %d\n", t_count))
  cat(sprintf("C: %d\n", c_count))
  cat(sprintf("G: %d\n", g_count))
  cat("\n")
}
```

Print a table of GC content for each sequence
```{r}
ncbi_ids <- c("Sequence 1", "Sequence 2", "Sequence 3")
sequences <- c(Sequences$Sequence)

# Create an empty data frame to store the results
gc_content_table <- data.frame(Sequence_ID = character(), GC_Content = numeric())

# Loop through each sequence ID and sequence
for (i in 1:length(ncbi_ids)) {
  seq_id <- ncbi_ids[i]
  seq <- sequences[i]
  
  # Count the number of Gs and Cs
  gc_count <- sum(grepl("G|C", strsplit(seq, "")[[1]]))
  
  # Calculate the percentage of GC content
  gc_content <- gc_count / nchar(seq) * 100
  
  # Add the results to the data frame
  gc_content_table[i, "Sequence_ID"] <- seq_id
  gc_content_table[i, "GC_Content %"] <- gc_content
}

# Print the final table
gc_content_table

```

[Borrelia burgdorferi](https://en.wikipedia.org/wiki/Borrelia_burgdorferi)
![Bacteria](./Bacteria.jfif)

#PART 2

In order to evaluate this mystery sequence I will test its similarity to annotated sequences found in the NCBI database using BLAST.
```{r}
isolate <- "GCCTGATGGAGGGGGATAACTACTGGAAACGGTAGCTAATACCGCATGAC
CTCGCAAGAGCAAAGTGGGGGACCTTAGGGCCTCACGCCATCGGATGAAC
CCAGATGGGATTAGCTAGTAGGTGGGGTAATGGCTCACCTAGGCGACGAT
CCCTAGCTGGTCTGAGAGGATGACCAGCCACACTGGAACTGAGACACGGT
CCAGACTCCTACGGGAGGCAGCAGTGGGGAATATTGCACAATGGGCGCAA" # make a vector containing the target sequence to be used in a BLAST
```
First I will search for similar sequences using BLAST.
```{r}
blast <- blastSequences(isolate, as='data.frame', timeout=200, hitListSize = 1) #this function took extremely long to run so I raised the timeout to 200 seconds and lowered the hit list size to 1, since I saw that there are some perfect matches in NCBI which should show up even with a minimal hit list
```
I will need to examine the structure of the blast data table to see what the results of the BLAST are.
```{r}
str(blast)
```
The information contained in the Hsp_Score and Hsp_evalue will inform us as to how well the mystery DNA matches the hits.
```{r}
blast$Hsp_score
blast$Hsp_evalue
```
Very high Hsp Scores and very low Hsp e values reflect that these hits match the sequence extremely well. 

I will then check on the identity of the hits that matched so well

```{r}
blast$Hit_def #check on the identity of the hit sequences
```

The mystery DNA sequence matches perfectly with the Yersinia pestis EV76-CN chromosome. Yersinia pestis is the bacteria that is responsible for the plague. In my professional opinion I do believe that due to the strength of the association between this DNA and known sequences from the Yersinia pestis genome, that this finding is a cause for concern, I suggest immediate action be taken to isolate and treat the individual that this DNA was isolated from.